worlds/pokemon_emerald/rules.py

```diff
@@ -1527,14 +1527,19 @@ def get_location(location: str):
     if world.options.dexsanity:
         for i in range(NUM_REAL_SPECIES):
             species = data.species[NATIONAL_ID_TO_SPECIES_ID[i + 1]]
+
+            if species.species_id in world.blacklisted_wilds:
+                continue
+
             set_rule(
                 get_location(f"Pokedex - {species.label}"),
                 lambda state, species_name=species.name: state.has(f"CATCH_{species_name}", world.player)
             )
 
         # Legendary hunt prevents Latios from being a wild spawn so the roamer
         # can be tracked, and also guarantees that the roamer is a Latios.
-        if world.options.goal == Goal.option_legendary_hunt:
+        if world.options.goal == Goal.option_legendary_hunt and \
+                data.constants["SPECIES_LATIOS"] not in world.blacklisted_wilds:
```

Comment:
Can I get a quick explanation on whether/how the client knows that it's now not supposed to check for Latios when it comes to actually firing the victory event? Couldn't work it out from a quick glance at the code. Otherwise, LGTM

Reply:
The client should still look for Latios even when the player blacklists it. Remember, legendary hunt cares about catching the roamer, not catching the species Latios. It just happens that the roamer is forced to be Latios, and Latios is automatically banned from wild encounters during legendary hunt.

- If goal _is not_ legendary hunt and Latios _is_ blacklisted, the Latios pokedex location doesn't exist and was skipped on line 1531.
- If goal _is not_ legendary hunt and Latios _is not_ blacklisted, the Latios pokedex location exists and had its access rule set on line 1534.
- If goal _is_ legendary hunt and Latios _is_ blacklisted, the Latios pokedex location doesn't exist and was skipped on line 1531. The roamer is still forced to be Latios.
- If goal _is_ legendary hunt and Latios _is not_ blacklisted, the Latios pokedex location exists and had its access rule set on line 1534. But the legendary hunt option manually blacklisted Latios from appearing in the wild, so the rule is impossible to satisfy, and the pokedex location looks unreachable. We need to change the rule for Latios pokedex into "can access roamer encounter", which is `EVENT_ENCOUNTER_LATIOS`.

So this change is working the same as it did, but if you blacklist Latios yourself, the location doesn't exist, so we don't ever want to try to set a rule on it.